### PR TITLE
Handle writing automatically

### PR DESCRIPTION
**Pull request by forslund**
Using decorator as suggested by @MatthewScholefield. The decorator sets
an internal attirbute so following calls to a decorated function will
not try to write data.

Apparently I didn't push this commit last week. Lets add it as well to keep the python interface the same.

### CODE REVIEW

File: msm/mycroft_skills_manager.py [hunk 1]
```diff
@@ -207,12 +226,16 @@ def update(self, skill=None, author=None):
...
 
+    @save_skills_data
     def apply(self, func, skills):
         """Run a function on all skills in parallel"""
 
         def run_item(skill):
             try:
-                func(skill)
+                if 'save_skills_data' in dir(func):
+                    func(skill, __msm_write=False)
```
Comment (MatthewScholefield) [inline msm/mycroft_skills_manager.py]: Does the `__msm_write=False` still refer to anything?

Comment (forslund) [reply msm/mycroft_skills_manager.py]: No, you're right. That's a remnant from my first approach. Removing

**Comment by forslund** [comment]
There, the code should be cleaner now.

**Review by MatthewScholefield — APPROVED** [review]
Looks good!

**Review by MatthewScholefield — CHANGES_REQUESTED** [review]
Just thought of an edge case where this could mess up:
If the function throws an exception I don't think it saves skills (probably desired anyways) but also might not reset the `saving_handled` variable. For example:
```python
try:
    msm.install(skill_name, skill_author)
except SkillNotFound:
    msm.install(skill_name)  # self.saving_handled already True but won't save
```

**Comment by forslund** [comment]
Ah good thinking. I'll think it through and do an update.

A try, finally might be enough to handle it.

**Review by MatthewScholefield — APPROVED** [review]

**Pull request merged**